src/main/ports.ts

```diff
@@ -0,0 +1,152 @@
+import { ConfigValue } from 'common/nodeConfig';
+import Node, { NodeConfig } from '../common/node';
+import { httpGet } from './httpReq';
+import { getNodes, getSetPortHasChanged } from './state/nodes';
+import { addNotification } from './state/notifications';
+import { NOTIFICATIONS } from './consts/notifications';
+
+export const getPodmanPortsForNode = (node: Node): ConfigValue[] => {
+  const portsArray = [] as ConfigValue[];
+  const { configTranslation } = node.spec;
+
+  if (!configTranslation) return portsArray;
+
+  // Extract default port values safely with optional chaining
+  const defaultHttpPort = configTranslation.httpPort?.defaultValue;
+  const defaultWebSocketsPort = configTranslation.webSocketsPort?.defaultValue;
+  const defaultP2pPorts = configTranslation.p2pPorts?.defaultValue;
+  const defaultP2pPortsUdp = configTranslation.p2pPortsUdp?.defaultValue;
+  let defaultP2pPortsTcp = configTranslation.p2pPortsTcp?.defaultValue;
+  const defaultEnginePort = configTranslation.enginePort?.defaultValue;
```

Comment:
Probably for the next version of this PR, unless you want to try now, we could try to make features like this as general as possible (not eth node specific). So a NodeSpec could look like...
```
    "someKindOfPort": {
      "displayName": "A node needs this port for something",
      "cliConfigPrefix": "--p2p-port=",
      "uiControl": {
        "type": "text"
      },
      "infoDescription": "Example value: 30303",
      "defaultValue": "30303",
      "documentation": "https://besu.hyperledger.org/public-networks/reference/cli/options#p2p-port"
... new fields...
      "portConfig" :{
           "isListening": true/false, // is this a port that the node is listening to or is it a port that the node will connect to another node at
           "udpAndTcp": 'both'/'udp'/'tcp',
           "shouldPortBeOpen" : true/false // this setting should only be here if "isListening" is set to true
       }
    },
```
Then we can just loop over all of a node's Spec for configTranslation.<key>.ports when checking ports and when setting the `-p <port>:<port>` for podman

Reply:
@jgresham Would it be okay if I looked into this after we start adding non-eth nodes (or does this also apply to L2 nodes?)? just because I'd also have to look up whether these individual ports are listening or not, and it works as is for the moment with the nodes that are currently supported.

If that's okay, this PR should be ready to be reviewed again and merged.

Reply:
Agree. I'll copy this conversation into a Linear ticket.